File: symfony-app/src/Entity/SuspiciousLogin.php
<?php

namespace App\Entity;

use App\Repository\SuspiciousLoginRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: SuspiciousLoginRepository::class)]
#[ORM\Table(name: 'suspicious_logins')]
class SuspiciousLogin
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(nullable: false, onDelete: 'CASCADE')]
    private User $user;

    #[ORM\Column(type: 'string', length: 64, nullable: true)]
    private ?string $ipAddress = null;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $userAgent = null;

    #[ORM\Column(type: 'string', length: 64, nullable: true)]
    private ?string $country = null;

    #[ORM\Column(type: 'string', length: 100)]
    private string $reason;

    #[ORM\Column(type: 'boolean', options: ['default' => true])]
    private bool $blocked = true;

    #[ORM\Column(type: 'boolean', options: ['default' => false])]
    private bool $notified = false;

    #[ORM\Column(type: 'datetime_immutable')]
    private \DateTimeImmutable $createdAt;

    public function __construct()
    {
        $this->createdAt = new \DateTimeImmutable();
    }

    public function getId(): ?int { return $this->id; }

    public function getUser(): User { return $this->user; }
    public function setUser(User $user): void { $this->user = $user; }

    public function getIpAddress(): ?string { return $this->ipAddress; }
    public function setIpAddress(?string $ipAddress): void { $this->ipAddress = $ipAddress; }

    public function getUserAgent(): ?string { return $this->userAgent; }
    public function setUserAgent(?string $userAgent): void { $this->userAgent = $userAgent; }

    public function getCountry(): ?string { return $this->country; }
    public function setCountry(?string $country): void { $this->country = $country; }

    public function getReason(): string { return $this->reason; }
    public function setReason(string $reason): void { $this->reason = $reason; }

    public function isBlocked(): bool { return $this->blocked; }
    public function setBlocked(bool $blocked): void { $this->blocked = $blocked; }

    public function isNotified(): bool { return $this->notified; }
    public function setNotified(bool $notified): void { $this->notified = $notified; }

    public function getCreatedAt(): \DateTimeImmutable { return $this->createdAt; }
    public function setCreatedAt(\DateTimeImmutable $createdAt): void { $this->createdAt = $createdAt; }
}

File: javafx-app/src/main/java/com/santea/model/SuspiciousLogin.java
package com.santea.model;

import java.time.LocalDateTime;

public class SuspiciousLogin {
    private Integer id;
    private User user;
    private String ipAddress;
    private String userAgent;
    private String country;
    private String reason;
    private Boolean blocked;
    private Boolean notified;
    private LocalDateTime createdAt;

    public SuspiciousLogin() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getIpAddress() {
        return ipAddress;
    }

    public void setIpAddress(String ipAddress) {
        this.ipAddress = ipAddress;
    }

    public String getUserAgent() {
        return userAgent;
    }

    public void setUserAgent(String userAgent) {
        this.userAgent = userAgent;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    public Boolean getBlocked() {
        return blocked;
    }

    public void setBlocked(Boolean blocked) {
        this.blocked = blocked;
    }

    public Boolean getNotified() {
        return notified;
    }

    public void setNotified(Boolean notified) {
        this.notified = notified;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}
